src/nimony/sem.nim

```diff
@@ -7121,6 +7121,57 @@ proc writeOutput(c: var SemContext; outfile: string) =
       toBuild: move c.toBuild,
       exportBuf: buildIndexExports(c))
 
+  # Update .2.deps.nif file that doesn't contain modules imported under `when false:`
+  # so that Hexer and following phases doesn't read such modules.
+  var deps = createTokenBuf(16)
+  deps.buildTree StmtsS, NoLineInfo:
+    if c.importedModules.len != 0:
+      let systemSymId = if {SkipSystem, IsSystem} * c.moduleFlags == {}:
+                          c.currentScope.tab[pool.strings.getOrIncl("system")][0].name
```

Comment:
Gross hack. And why does system to be special cased?

Reply:
In the first commit in this PR, sem.nim didn't exclude system module from `.1.deps.nif` and that caused compile errors as nimsem compiled system module without `--isSystem`.
`src/nimony/deps.nim` supposes system module is not in `.deps.nif` and system module is handled differently from normal modules.

Current nimsem writes `.2.deps.nif` and they are used to generate final Makefile that calls Hexer and later phases that doesn't handle system module differently.
So sem.nim can include system module in `.2.deps.nif` now.

Excluding system module from `.deps.nif` makes file size smaller.